website/react-magma-docs/src/pages/api/transition.mdx

```diff
@@ -0,0 +1,599 @@
+          transitions.nudgeRight
+        }
+        onChange={() =>
+          setTransitions({ ...transitions, slideLeft: !transitions.slideLeft })
+        }
+      />
+      <Checkbox
+        labelText="nudgeTop"
+        checked={transitions.nudgeTop}
+        disabled={
+          transitions.collapse |
+          transitions.slideTop |
+          transitions.slideBottom |
+          transitions.slideLeft |
+          transitions.slideRight |
+          transitions.nudgeBottom
+        }
+        onChange={() =>
+          setTransitions({ ...transitions, nudgeTop: !transitions.nudgeTop })
+        }
+      />
+      <Checkbox
+        labelText="nudgeBottom"
+        checked={transitions.nudgeBottom}
+        disabled={
+          transitions.collapse |
+          transitions.slideTop |
+          transitions.slideBottom |
+          transitions.slideLeft |
+          transitions.slideRight |
+          transitions.nudgeTop
+        }
+        onChange={() =>
+          setTransitions({
+            ...transitions,
+            nudgeBottom: !transitions.nudgeBottom,
+          })
+        }
+      />
+      <Checkbox
+        labelText="nudgeRight"
+        checked={transitions.nudgeRight}
+        disabled={
+          transitions.collapse |
+          transitions.slideTop |
+          transitions.slideBottom |
+          transitions.slideLeft |
+          transitions.slideRight |
+          transitions.nudgeLeft
+        }
+        onChange={() =>
+          setTransitions({
+            ...transitions,
+            nudgeRight: !transitions.nudgeRight,
+          })
+        }
+      />
+      <Checkbox
+        labelText="nudgeLeft"
+        checked={transitions.nudgeLeft}
+        disabled={
+          transitions.collapse |
+          transitions.slideTop |
+          transitions.slideBottom |
+          transitions.slideLeft |
+          transitions.slideRight |
+          transitions.nudgeRight
+        }
+        onChange={() =>
+          setTransitions({ ...transitions, nudgeLeft: !transitions.nudgeLeft })
+        }
+      />
+
+      <Preview />
+
+      <Transition in={isOpen} unmountOnExit={unmountOnExit} {...transitions}>
+        <Box />
+      </Transition>
+
+      <Button onClick={() => setIsOpen(!isOpen)}>
+        {isOpen ? 'hide' : 'show'}
+      </Button>
+    </div>
+  );
+}
+```
+
+## Basic Usage
+
+### Fade
+
+```typescript
+import React from 'react';
+import { Transition } from 'react-magma-dom';
+
+export function Example() {
+  const [isOpen, setIsOpen] = React.useState(false);
+  return (
+    <div>
+      <Transition in={isOpen} fade style={{ overflow: 'hidden' }}>
+        <div
+          style={{
+            height: '250px',
+            width: '250px',
+            backgroundColor: '#bada55',
+          }}
+        />
+      </Transition>
+      <Button onClick={() => setIsOpen(!isOpen)}>
+        {isOpen ? 'hide' : 'show'}
+      </Button>
+    </div>
+  );
+}
+```
+
+### NudgeTop
```

Comment:
These examples behave a little oddly and not consistently with each other. The div seems to be growing from just one side when transitioning in.  The NudgeBottom example is always on the screen and the NudgeRight example doesn't show up at all.

I also think it would be helpful to add an explanation of what each transition type actually does.